awscli/customizations/paginate.py

```diff
@@ -337,7 +338,15 @@ def add_to_parser(self, parser):
         parser.add_argument(self.cli_name, dest=self.py_name,
                             type=self.type_map[self._parse_type])
 
+    def _check_and_warn_negative_max_items(self, value):
+        if int(value) <= 0:
+            uni_print(
+                "warning: Non-positive values for --max-items are unsupported and may yield undefined behavior.\n",
```

Comment:
Nit: Saying they are unsupported seems that they would raise an error; they are allowed but will likely result in nonsense.
 
```suggestion
                "warning: Non-positive values for --max-items may result in undefined behavior.\n",
```